### PR TITLE
Add additional filter options to rally list races

### PR DESCRIPTION
**Pull request by ebadyano**
- Added extra files to display for listing races ( `ES Version`, `Revision` and `Rally Version`).
- Removed `Track Parameters` from being displayed, as for some races (e.g. logs) it's very long and makes it hard to understand the output. 
- Added new filters for list races: `--track`, `--name` and `--to-date`, `--from-date`

### CODE REVIEW

File: esrally/rally.py [hunk 1]
```diff
@@ -144,6 +154,28 @@ def add_track_source(subparser):
         help="Limit the number of search results for recent races (default: 10).",
         default=10,
     )
+    list_parser.add_argument(
```
Comment (michaelbaamonde) [inline esrally/rally.py:157]: With the way things are structured right now, all `list` commands will accept these new race-specific arguments, but can't do anything useful with them. It's not a huge deal, but seeing them in the `--help` output for something like `esrally list tracks --help` could add some confusion.

That said, there's already precedent for this. `--limit` only really applies to races, but you can pass it to any other `list` subcommand.

Could you perhaps change the help text for the new arguments to indicate that they're intended to be used for races only? Something like `Show only races for this track` in this case, for example.

**Review by michaelbaamonde — CHANGES_REQUESTED** [review]
I tested out the basic functionality with both a local and remote metrics store and it worked well! This is a very nice improvement, thanks.

In addition to my comments on adjusting the help text, I think we should add a unit test for the new filtering logic contained in `metrics.py`.

**Comment by ebadyano** [comment]
@michaelbaamonde thank you for your review. I added unit tests for the new filter logic. Thank you!

**Review by michaelbaamonde — APPROVED** [review]
LGTM!

**Pull request merged**